data_structures/src/chain.rs

```diff
@@ -50,17 +50,24 @@ pub struct ConsensusConstants {
     pub reputation_punishment: f64,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
-pub struct Checkpoint {
-    pub number: u32,
-    pub hash: Vec<u8>,
+/// Checkpoint beacon structure
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct CheckpointBeacon {
+    /// The serial number for an epoch
+    pub checkpoint: Epoch,
+    /// The 256-bit hash of the previous block header
+    pub hash_prev_block: Vec<u8>,
```

Comment:
Just a suggestion, but I would love to see more elaborated documentation, currently the auto-generated docs for this module looks like this:
![image](https://user-images.githubusercontent.com/83847/48939184-b717af80-ef13-11e8-918c-4801369d62dd.png)
I would love to see described what the ChainInfo is, why it is needed and where it is used, the "serial number for an epoch" is a bit vague IMO, the documentation already creates a link for that type so I would add something like why the checkpoint is of type Epoch or why it is necessary. Take for example how the documentation for the config looks like:
![image](https://user-images.githubusercontent.com/83847/48939411-6a80a400-ef14-11e8-94c4-4665b3d5ce46.png)